bson/value_reader.go

```diff
@@ -38,6 +53,33 @@ type valueReader struct {
 	frame int64
 }
 
+func getDocumentReader(r io.Reader) *valueReader {
+	vr := vrPool.Get().(*valueReader)
+
+	vr.offset = 0
+	vr.frame = 0
+
+	vr.stack = vr.stack[:1]
+	vr.stack[0].mode = mTopLevel
```

Comment:
What do you think of initializing other fields in `stack` to gain greater peace of mind?
e.g.
```
vr.stack[0] = vrState{mode: mTopLevel}
```